app/components/canvas/canvas.js

```diff
@@ -78,6 +78,52 @@ class CanvasController {
 
   setupMouseEventHandlers_() {
     this.canvas_
+        .on('mousedown', event => {
+          let canvasOffset = this.canvas_.offset();
+          let x = (event.pageX - canvasOffset.left) / this.scale_;
+          let y = (event.pageY - canvasOffset.top) / this.scale_;
+          let matrices = [];
+          // TODO(alockwood): select clips and/or groups in addition to paths?
+          let toggleSelectedPath_ = layer => {
+            if (layer instanceof LayerGroup) {
+              let transformMatrices = this.createTransformMatrices_(layer);
+              matrices.splice(matrices.length, 0, ...transformMatrices);
+              let result = layer.layers.some(layer => toggleSelectedPath_(layer));
+              matrices.splice(-transformMatrices.length, transformMatrices.length);
+              return result;
+            } else if (layer instanceof PathLayer && layer.pathData) {
+              let shouldUpdateSelection = false;
+              let reversedMatrices = Array.from(matrices).reverse();
+              let pointTransformerFn = p => this.transformPoint_(p, reversedMatrices);
+              if (layer.fillColor) {
+                shouldUpdateSelection = layer.pathData.isFillSelected({x, y}, pointTransformerFn);
+              } else if (layer.strokeColor) {
+                shouldUpdateSelection =
+                    layer.pathData.isStrokeSelected({x, y}, pointTransformerFn, layer.strokeWidth);
+              }
+              if (shouldUpdateSelection) {
+                if (event.metaKey || event.shiftKey) {
+                  this.studioState_.toggleSelected(layer);
```

Comment:
here and below we probably need to:

`layer = this.artwork.findLayerById(layer.id);`

to ensure that the root layer is selected instead of the rendered art layer (in the case where you're inside an animation block)